src/lib/Form/Type/Content/Draft/ContentCreateType.php

```diff
@@ -119,6 +120,21 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                         'content_draft_create_type.create',
                 ]
             );
+
+        $builder->addEventListener(
+            FormEvents::PRE_SUBMIT,
+            function (PreSubmitEvent $event): void {
+                $location = $this->locationService->loadLocation(
+                    (int)$event->getData()['parent_location']
+                );
+                $form = $event->getForm();
+                $opts = $form->get('content_type')->getConfig()->getOptions();
+                $opts['choice_loader'] = $this->contentCreateContentTypeChoiceLoader
+                    ->setTargetLocation($location);
```

Comment:
Instead of acquiring the ChoiceLoader object during the event from field options, store the ChoiceLoader in a variable and pass it into the function (`use ($choiceLoader)` or akin to that).